config/routes.rb

```diff
@@ -46,7 +46,8 @@
   resource :core_induction_programme, controller: :core_induction_programme, only: :show, as: "cip", path: "/core_induction_programme" do
     resource :years, controller: "core_induction_programme/years", only: :show, path: "/:year_id" do
       resource :modules, controller: "core_induction_programme/modules", only: :show, path: "/:module_id" do
-        resource :lessons, controller: "core_induction_programme/lessons", only: :show, path: "/:lesson_id"
+        resource :lessons, controller: "core_induction_programme/lessons", only: %i[show create edit update], path: "/:lesson_id"
```

Comment:
You could try and define this as a member route: resource :lessons, controller: "core_induction_programme/lessons", only: %i[show create edit update], path: "/:lesson_id" do
  member do
    put 'edit'
  end
end
https://guides.rubyonrails.org/routing.html#adding-member-routes and it's more like the rails way.